tox.ini

```diff
@@ -42,7 +43,7 @@ commands =
 deps = pre-commit
 skip_install = true
 commands =
-    pre-commit run --all-files --show-diff-on-failure
+    pre-commit run --all-files --show-diff-on-failure flake8
```

Comment:
Would renaming the style stage be a better way to handle this instead of specifying the flake stage explicitly and making a new bandit stage?

Reply:
I was thinking about that earlier, but went this way because `--all-files` was causing bandit to hit files it should ignore.  Let me check and see if I can specify an exclude pattern for bandit that works inside of precommit though...

Reply:
Good idea -- turns out I had already excluded the problematic tests directory. 

fba0d1fe27df19e8498c4a143be2f7a1fc8e1d0b . 

"style" may not be the 100% perfect name for the stage now, but I still thinks It's a pretty good name, so I left it. 

`pre-commit run --all-files` doesn't seem to include submodules (so we'd need to do something else to scan plugins -- but, I'm borderline on whether plugin checks should be in this repo anyway).

Reply:
I think pre-commits for submodules should happen in that module's repository. While that may be a little more messy, it allows more granular control of submodule testing and verification in the situations that may require a submodule to exclude additional files, ignore different rules, or etc.

Reply:
I 95% agree with that.  It's super annoying when a pipeline fails because of whitespace in a different repo haha.  

The 5% is probably what you meant by a "little more messy" --  maintaining and ensuring that all supported plugins are setup to run the right checks is extra effort. Any time we decide we want to add or tweak something in CI we'd probably want to do it for most of the included plugins. Things should be fairly standard though so it's probably just copy and paste.